common/thorhelper/thorcommon.hpp

```diff
@@ -29,7 +29,7 @@
 #include "thorhelper.hpp"
 #include "thorxmlwrite.hpp"
 
-static unsigned const defaultDaliResultOutputMax = 2000; // MB
+static unsigned const defaultDaliResultOutputMax = 100; // MB
```

Comment:
What happens when the limit is specified higher in a #option?

Reply:
It fires an exception, e.g.:
                throw MakeStringException(0, "Dali result outputs are restricted to a maximum of %d MB, the current limit is %d MB. A huge dali result usually indicates the ECL needs altering.", defaultDaliResultOutputMax, defaultDaliResultLimit);

Reply:
At compile time? Or at runtime?

(checking the value at compile time would be friendlier, if practical)

Reply:
The existing checks are at runtime in hthor/roxie/thor.
I think still need runtime check, because limit may come from ECL MAXSIZE(<expr>), which could be runtime dependent.
But I've added a compile time check also.

@ghalliday - can you check, and if constant in eclhelper.hpp is in a suitable place?

Reply:
The changes look fine.  I'm not sure where that constant should go.  It is a bit out of place in eclhelper.hpp.  I suspect the original place might be better - hqlcpp does include other things from thorhelper already.  I'm not sure it is worth changing though.
I have one example (kel11.eclxml) that sets that options, and it sets it to 1000.  Thant means this change is likely to break that query.

Reply:
We are deliberately breaking that query... before that query breaks Dali

Reply:
yes, just warning that it looks like it will be hit.
@jakesmith needs squashing

Reply:
Do you know who we need to warn?

Reply:
Apart from redbook, I can send an email out to ECLUsers@risk.lexisnexis.com once this is merged/tagged into a build?